lib/moog.js

```diff
@@ -192,17 +192,24 @@ module.exports = function(options) {
           }
           if (properties.group) {
             const groups = klona(that[`${cascade}Groups`]);
-            for (const value of Object.values(properties.group)) {
+            for (const [ key, value ] of Object.entries(properties.group)) {
               for (const field of value.fields || []) {
-                for (const value of Object.values(groups)) {
-                  if (value.fields) {
-                    if (value.fields.includes(field)) {
-                      value.fields = value.fields.filter(_field => _field !== field);
+                // Remove fields from existing groups if they're added to a new
+                // group.
+                for (const val of Object.values(groups)) {
+                  if (val.fields) {
+                    if (val.fields.includes(field)) {
+                      val.fields = val.fields.filter(_field => _field !== field);
                     }
                   }
                 }
               }
+              // Combine groups of the same name
+              if (groups[key] && Array.isArray(groups[key].fields)) {
+                value.fields = groups[key].fields.concat(value.fields);
```

Comment:
Since we're giving the new groups priority when spreading them below, let's combine on the new group's fields rather than try to be more fancy about it.